pkg/GAPJulia/JuliaInterface/read.g

```diff
@@ -6,23 +6,22 @@
 ReadPackage( "JuliaInterface", "gap/JuliaInterface.gi");
 ReadPackage( "JuliaInterface", "gap/BindCFunction.gi" );
 
-dirs:= DirectoriesPackageLibrary( "JuliaInterface", "julia" );
-
-JuliaIncludeFile( Filename( dirs, "gaptypes.jl" ) );
-
-dirs:= DirectoriesPackageLibrary( "JuliaInterface", "../LibGAP.jl/src" );
-
 if not IsBound( JULIAINTERNAL_LOADED_FROM_JULIA ) then
-    JuliaEvalString( Concatenation( "Base.include( Main,\"", Filename( dirs, "initialization.jl" ), "\")" ) );
+    JuliaEvalString( "__IS_LOADED_FROM_GAP = true" );
+    JuliaEvalString( "Base.MainInclude.include(Base.find_package(\"GAP\"))" );
 fi;
 
-JuliaEvalString( Concatenation( "Base.include( Main.GAP,\"", Filename( dirs, "libgap.jl" ), "\")" ) );
+dirs_julia := DirectoriesPackageLibrary( "JuliaInterface", "julia" );
+
+JuliaIncludeFile( Filename( dirs_julia, "gaptypes.jl" ) );
+
+JuliaEvalString( Concatenation( "__JULIAGAPMODULE.include( \"", Filename( dirs_julia, "libgap.jl" ), "\")" ) );
 
 _JULIAINTERFACE_INTERNAL_INIT();
 
+Julia!.storage.GAP := _WrapJuliaModule( "GAP", _JuliaGetGlobalVariable( "__JULIAGAPMODULE" ) );
```

Comment:
Could you explain what this line is about? Why do we need it, and why does it expose `__JULIAGAPMODULE` under the name "GAP", instead of putting the module "GAP" there?
Perhaps a comment could then be added to the file with that explanation?

Reply:
I added a comment.

Reply:
Indeed, you even added two comments. That helped a lot, thanks!